### PR TITLE
Updated animist.js to use class for source links, fixing unique ID is…

### PR DESCRIPTION
**Pull request by biplab-sutradhar**
Updates animist.js to use classes for source links, resolving issue #145.

### CODE REVIEW

**Comment by tdhock** [comment]
Hi, thanks for your contribution! 
Please also edit the test case, grep a_source_href in testthat/* files

**Comment by biplab-sutradhar** [comment]
@tdhock I've edited the test case, thank you for your advice.

**Comment by tdhock** [comment]
for PRs like this one that would close an issue, please put the following text in the first PR comment, which tells github to close the issue when the PR is merged.

Closes #145

**Comment by tdhock** [comment]
I invited you to animint org, so after you accept that, you will have write permission to this repo.
please create future PRs using branches in this repo, instead of your fork.
(and please avoid pushing to master)

**Comment by biplab-sutradhar** [comment]
Thank you, tdhock!

I've accepted the invitation to join the animint organization, and I’ll create future PRs using branches also make sure not to push directly to master.

**Comment by biplab-sutradhar** [comment]
should I create a new branch here before commit

**Comment by tdhock** [comment]
please just submit the changes to this branch/pr and then I will merge.

**Comment by biplab-sutradhar** [comment]
@tdhock I have updated news.md, description and in this issue one error coming during testing "GitHub API error (403): Must have admin rights to the repository." This issue is most likely related to the environment variable.

**Comment by tdhock** [comment]
that test failure always happens in PRs from branches in forks, because that test needs permission which is only available using token in this repo.

**Comment by tdhock** [comment]
> I have updated news.md, description 

I don't see these updates, did you forget to push?

**Comment by biplab-sutradhar** [comment]
@tdhock Sorry for my misunderstanding. Should I push my code to the pr branch, or to the bugfix/issue-123 branch (which I created for issue #154) If I create a new branch I need to make a new PR as well?

**Comment by tdhock** [comment]
in this PR you should describe what changes you made in NEWS, and increment the version in DESCRIPTION
to update this PR you need to push to biplab-sutradhar:bugfix/issue-123

**Comment by biplab-sutradhar** [comment]
Updated NEWS file to reflect changes made in resolving issue #154. 
 Added name and role as a contributor, incremented version number in DESCRIPTION based on the current date

**Comment by biplab-sutradhar** [comment]
@tdhock extra newline issue was not showing in vscode hence I directly commit in github

**Comment by tdhock** [comment]
please click "Files changed" tab at the top of this PR page, which leads to https://github.com/animint/animint2/pull/154/files which shows that there is still no change in DESCRIPTION and NEWS.
please make sure to preview your changed in Files changed tab before asking for review, because that is how other people review your PRs.

**Comment by biplab-sutradhar** [comment]
@tdhock Sorry for the misunderstanding. I have updated.

File: NEWS.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+# Changes in version 2024.10.10 (PR#145)
```
Comment (tdhock) [inline NEWS.md]: please change to PR#154 (145 is the issue)

**Review by tdhock — CHANGES_REQUESTED** [review]

File: NEWS.md (hunk 1 above)
Comment (biplab-sutradhar) [reply NEWS.md]: Yes, I changed to PR#154 on my last commit.

File: NEWS.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
+# Changes in version 2024.10.10 (PR#154)
+
+  - Updated `animist.js` to use classes for source links, resolving issue #154.
```
Comment (tdhock) [inline NEWS.md:3]: now the issue number is incorrect please revise

**Review by tdhock — CHANGES_REQUESTED** [review]

File: NEWS.md (hunk 2 above)
Comment (biplab-sutradhar) [reply NEWS.md]: https://github.com/animint/animint2/pull/154/commits/703d0656ad6b89d0f8ee52aaa175b2df7be8134c

this commit was correct, but I have updated to my last commit. Sorry for the misunderstanding.

**Comment by tdhock** [comment]
ok thanks

**Pull request merged**